react-components/src/App.tsx

```diff
@@ -0,0 +1,60 @@
+import { Fragment, useMemo } from 'react'
+import s from './App.scss'
+import SearchBar from './components/SearchBar/SearchBar'
+import WeatherCard from './components/WeatherCard/WeatherCard'
+import { getWeatherData } from './API/weatherAPI'
+
+const WeatherCards: React.FC = () => {
+    const weatherData = useMemo(getWeatherData, [])
+
+    return (
+        <Fragment>
+            {weatherData.map(([name, data]) => {
+                const { temperatureLimit, description, imageUrl } = data
+                const { probability = -1, type, iconUrl } = data.chanceTo || {}
+
+                const additionalInfo = data.chanceTo && (
```

Comment:
I like it. This is a good practice. When you break down into separated variables the logic and at the end you compound them up.